src/HLSLTree.cpp

```diff
@@ -382,6 +382,40 @@ bool HLSLTree::GetExpressionValue(HLSLExpression * expression, int & value)
     return false;
 }
 
+bool HLSLTree::NeedsFunction(const char* name)
+{
+    // Early out
+    if (!GetContainsString(name))
+        return false;
+
+    struct NeedsFunctionVisitor: HLSLTreeVisitor
```

Comment:
I agree, it's not a perf issue for us either. You are right it doesn't help the most common case. Terminating early is easy, though. Just override the statement traversal.